File: csharp/exceptions1/Program.cs
using System;

namespace Exceptions
{

    class Program
    {
        public static void Main(string[] args)
        {
            Car car1 = new Car();
            try
            {
                //car1 = null;
                Console.WriteLine("Make:{0} Model{1}", car1.Make, car1.Model);

                car1 = new Car();
                car1.Make = "Hundai";
                car1.Model = "Getz";
                Console.WriteLine("Make:{0} Model{1}", car1.Make, car1.Model);
            }
            //catch (System.NullReferenceException e)
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            finally{
                Console.WriteLine("This block will be executed whether or not there is an exception");
            }

        }
    }

    class Car
    {
        public string Make { get; set; }
        public string Model { get; set; }
    }

}

File: java/exceptions1/Program.java
import java.util.*;
import java.io.*;
import java.lang.*;

public class Program {

    public static void main(String[] args) {
        try {
            Car car1 =new Car(); // java.lang.NullPointerException
            displayCar(car1);
        } catch (NullPointerException exp) {
            System.out.println("Null exception-->" + exp.getMessage());
        } catch (Exception exp) {
            System.out.println("exception-->" + exp.getMessage());
        } finally {
            System.out.println("This block is executed whether or not there is an exceptionn");
        }
    }

    public static void displayCar(Car car) throws NullPointerException, Exception {
            if((car!=null)&&(car.Make==null || car.Model==null)){
                throw new Exception("Car Make or Model cannot be null"); // Custom exception
            }
            System.out.println("Make:" + car.Make + "\tModel:" + car.Model);
       }

}

class Car {
    public String Make;
    public String Model;
}
